File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = prices[0]
        max_profit = 0

        for i in range(len(prices)):
            price = prices[i]
            min_price = min(min_price, price)
            profit = price - min_price
            max_profit = max(max_profit, profit)

        return max_profit

File: Data Structures & Algorithms/max-water-container/submission-1.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int max = 0;

        for (int i = 0; i < heights.Length; i++)
        {
            for (int j = i + 1; j < heights.Length; j++)
            {
                int min = Math.Min(heights[i], heights[j]);
                int product = (j - i) * min;
                max = Math.Max(max, product);
            }
        }

        return max;
    }
}
